[BUG][CMake] Typo in `CMakeLists.txt`, export set is always empty

`src/CMakeLists.txt` [has](https://github.com/hpc-io/vfd-gds/blob/master/src/CMakeLists.txt#L104C1-L109C4)
```cmake
  export(
    TARGETS
      ${HDF5_VFD_GDS_EXPORTED_LIBS}
    FILE
      ${HDF5_VFD_GDS_EXPORTED_TARGETS}.cmake
  )
```
but `HDF5_VFD_GDS_EXPORTED_LIBS` is always empty which causes
```sh
CMake Error in some/parent/project/CMakeLists.txt:
  export called with target "SomeParentProject" which requires target "hdf5_vfd_gds"
  that is not in any export set.
```
This is [because](https://github.com/hpc-io/vfd-gds/blob/master/src/CMakeLists.txt#L54C1-L54C88):
```cmake
set(HDF5_VFD_GDS_EXPORTED_LIBS hdf5_vfd_gds ${HDF5_VFD_GDS_EXPORTED_LIBS} PARENT_SCOPE)
```
only sets the new value in the _parent scope_, which does _not_ trickle down to the child scope (thanks CMake for this brain-damaged design...).

Changing this to
```cmake
set(HDF5_VFD_GDS_EXPORTED_LIBS hdf5_vfd_gds ${HDF5_VFD_GDS_EXPORTED_LIBS})
set(HDF5_VFD_GDS_EXPORTED_LIBS ${HDF5_VFD_GDS_EXPORTED_LIBS} PARENT_SCOPE)
```
fixes the problem

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -51,7 +51,8 @@ if(HDF5_VFD_GDS_ENABLE_COVERAGE)
   set_coverage_flags(hdf5_vfd_gds)
 endif()
 
-set(HDF5_VFD_GDS_EXPORTED_LIBS hdf5_vfd_gds ${HDF5_VFD_GDS_EXPORTED_LIBS} PARENT_SCOPE)
+set(HDF5_VFD_GDS_EXPORTED_LIBS hdf5_vfd_gds ${HDF5_VFD_GDS_EXPORTED_LIBS})
+set(HDF5_VFD_GDS_EXPORTED_LIBS ${HDF5_VFD_GDS_EXPORTED_LIBS} PARENT_SCOPE)
 
 #-----------------------------------------------------------------------------
 # Specify project header files to be installed

